app/boards/shields/leeloo_micro/leeloo_micro_common.dtsi

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <dt-bindings/zmk/matrix_transform.h>
+
+/ {
+    chosen {
+        zephyr,display = &oled;
+        zmk,kscan = &kscan0;
+        zmk,matrix_transform = &default_transform;
+    };
+
+    default_transform: keymap_transform_0 {
+        compatible = "zmk,matrix-transform";
+        columns = <10>;
+        rows = <4>;
+//
+// | SW1  | SW2  | SW3  | SW4  | SW5  |                 | SW15 | SW4  | SW3  | SW2  | SW1  |
+// | SW6  | SW7  | SW8  | SW9  | SW10 |                 | SW10 | SW9  | SW8  | SW7  | SW6  |
+// | SW11 | SW12 | SW13 | SW14 | SW15 | SW20 |   | SW20 | SW15 | SW14 | SW13 | SW12 | SW11 |
+//               | SW16 | SW17 | SW18 | SW19 |   | SW19 | SW18 | SW17 | SW16 |
+        map = <
+RC(0,0) RC(0,1) RC(0,2) RC(0,3) RC(0,4)                 RC(0,5) RC(0,6) RC(0,7) RC(0,8) RC(0,9)
+RC(1,0) RC(1,1) RC(1,2) RC(1,3) RC(1,4)                 RC(1,5) RC(1,6) RC(1,7) RC(1,8) RC(1,9)
+RC(2,0) RC(2,1) RC(2,2) RC(2,3) RC(2,4) RC(3,4) RC(3,5) RC(2,5) RC(2,6) RC(2,7) RC(2,8) RC(2,9)
+                RC(3,0) RC(3,1) RC(3,2) RC(3,3) RC(3,6) RC(3,7) RC(3,8) RC(3,9)
+        >;
+    };
+
+    kscan0: kscan {
+        compatible = "zmk,kscan-gpio-matrix";
+        label = "KSCAN";
+
+        diode-direction = "col2row";
+
+        row-gpios
+            = <&pro_micro 5 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+            , <&pro_micro 6 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+            , <&pro_micro 7 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+            , <&pro_micro 8 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+            ;
+    };
+
+    left_encoder: left_encoder {
+        compatible = "alps,ec11";
+        label = "LEFT_ENCODER";
+        a-gpios = <&pro_micro 21 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+        b-gpios = <&pro_micro 20 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+        resolution = <2>;
```

Comment:
I believe these encoders should be refactored for the changes noted in https://zmk.dev/blog/2023/06/18/encoder-refactors

Reply:
@caksoylar, I believe I have the new encoder configuration setup properly.  It seems to work well based on my tests.

Reply:
Hi @caksoylar,

I was able to do some more research on the rotary encoders that I'm using.  Based on the data sheet, the pulses are 15 and the detents are 30.  I've adjusted configuration items in the .dtsi files of Leeloo and Leeloo-Micro and they work well.  But, not at a reasonable resolution.  For each detent, there is an increase of 8 (i.e., 0, 8, 16, etc).  When I change the pulses to 120, it changes the resolution to 2 (i.e., 0, 2, 4, etc).  This makes each detent feel more predictable.  I've set both .dtsi files to 120/30.

Thank you!

:)